momentGW/pbc/tda.py

```diff
@@ -242,12 +307,61 @@ def build_se_moments(self, moments_dd):
                         Lp = self.integrals.Lpx[kp, kx][:, :, x]
                         subscript = f"P{pchar},Q{qchar},PQ->{pqchar}"
                         eta[kp, q][x, n] += util.einsum(subscript, Lp, Lp.conj(), eta_aux)
+                        if self.fc and q == 0:
+                            original = eta[kp, q][x, n]
+
+                            eta[kp, q][x, n] += (2 / np.pi) * (q0) * eta_head[kp, n] * original
+
+                            wing_tmp = util.einsum("Pp,P->p", Lp, eta_wings[kp, n])
+                            wing_tmp = wing_tmp.real * 2
+                            wing_tmp *= -(np.sqrt(cell_vol / (4 * (np.pi**3))) * q0**2)
+
+                            eta[kp, q][x, n] += util.einsum("p,pq->pq", wing_tmp, original)
 
         # Construct the self-energy moments
         moments_occ, moments_vir = self.convolve(eta)
 
         return moments_occ, moments_vir
 
+    def build_pert_term(self, qpt):
+        """
+        Compute 1/sqrt(Ω) * ⟨Ψ_{ik}|e^{iqr}|Ψ_{ak-q}⟩ at q-point index
+        q using perturbation theory.
```

Comment:
add params and return sections, also use the NumPy convention for the maths: https://numpydoc.readthedocs.io/en/latest/format.html#notes